### DISCUSSION TITLE
Port forwarding rework

### DISCUSSION THREAD

**Discussion by qdm12** [qdm12/gluetun#542] [Q&A] [answered]
I need your help testing out `qmcgaw/gluetun:pf-rework` before I merge in the main image.

It reworks a lot of PIA's port forwarding code to unblock me in implementing Wireguard.

**Comment by ghost** [comment, marked as answer]
Looks like it's working nicely so far. Port is pulled, opened and working in torrent clients at least. Anything specific you want tested/checked?

**Reply by qdm12** [reply]
Nice 👍 That was quicker than I thought 😄 

One more little thing (maybe you can learn that too!), is to restart openvpn (I assume your container is named `gluetun`):

```sh
docker exec -it gluetun /bin/sh
apk add curl
curl -X PUT -d '{"status":"stopped"}' http://localhost:8000/v1/openvpn/status
curl -X PUT -d '{"status":"running"}' http://localhost:8000/v1/openvpn/status
exit
```

What do you get in your logs?

EDIT: changed `--body` to `-d`

**Reply by ghost** [reply]
Very nice! Changed --body to --data and I can see in the logs that the port forwarding stops and the file gets removed, and then after connecting again everything is set up the same way. Nothing of concern to report from the logs tbh.

**Reply by qdm12** [reply]
Nice perfect, thanks for the testing!

I'll merge it soon.

And yes sorry for the `--body` it's `-d` indeed!

Have a great day/evening!
